core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,659 @@
+    # feedback corresponds to.
+    platform = datastore_services.StringProperty(
+        required=True, indexed=True, choices=PLATFORM_CHOICES)
+    # The ID of the user that scrubbed this report, if it has been scrubbed.
+    scrubbed_by = datastore_services.StringProperty(
+        required=False, indexed=True)
+    # Unique ID for the ticket this report is assigned to (see
+    # AppFeedbackReportTicketModel for how this is constructed).
+    ticket_id = datastore_services.TextProperty(default=None, indexed=False)
+    # Datetime in UTC of when the report was submitted by the user on their
+    # device. This may be much earlier than the model entity's creation date if
+    # the report was locally cached for a long time on an Android device.
+    submitted_on = datastore_services.DateTimeProperty(
+        required=True, indexed=True)
+    # The type of feedback for this report; choices are not specified as
+    # iterations of the report structure may introduce new types and we cannot
+    # rely on the backend updates to fully sync with the frontend report
+    # updates.
+    report_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The category that this feedback is for.
+    category = datastore_services.TextProperty(required=True, indexed=False)
+    # The version of the app; on Android this is the package version name (e.g.
+    # 1.0-release-arm...) and on web this is the release version (e.g. 3.0.8).
+    platform_version = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The user's country locale represented as a ISO-3166 code; the locale is
+    # determined by the user's Android device settings.
+    device_country_locale_code = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The entry point location that the user is accessing the feedback report
+    # from on both web & Android devices. On Android, this could be
+    # navigation_drawer, lesson_player, revision_card, or crash.
+    entry_point = datastore_services.StringProperty(required=True, indexed=True)
+    # Additional topic / story / exploration IDs that may be collected depending
+    # on  the entry_point used to send the report; lesson player will have
+    # topic_id, story_id, and exploration_id, while revision card while have
+    # topic_id and subtopic_id.
+    entry_point_topic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_story_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_exploration_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+    entry_point_subtopic_id = datastore_services.StringProperty(
+        required=False, indexed=True)
+
+    # The text language on Oppia set by the user in its ISO-639 language code;
+    # this is set by the user in Oppia's app preferences. Current languages on
+    # Android can be: EN, FR, HI, or ZH.
```

Comment:
Descriptions need to be up-to-date, and chances are that if the Android language set changes, this will not get updated. Suggest writing it to make it more future-proof. Ditto below.

Reply:
Ack, makes sense. Removing the specific languages since I don't think these are actually that helpful to list them out here -- validation is happening elsewhere.